File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>
using namespace std;

int m, x, s;
string cmd;

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	cin >> m;

	while (m--) {
		cin >> cmd;
		if (cmd == "all")
			s = ~(1 << 21);
		else if (cmd == "empty")
			s = 0;
		else {
			cin >> x;
			if (cmd == "add")
				s |= 1 << x;
			else if (cmd == "remove")
				s &= ~(1 << x);
			else if (cmd == "check")
				cout << (s & (1 << x) ? 1 : 0) << '\n';
			else
				s ^= 1 << x;
		}
	}
	

	return 0;
}

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		int m = Integer.parseInt(br.readLine());
		Set<Integer> set = new HashSet<>();

		while (m-- > 0) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			String s = st.nextToken();

			if (s.equals("all")) {
				for (int i = 1; i <= 20; i++) {
					set.add(i);
				}
			} else if (s.equals("empty")) {
				set = new HashSet<>();
			} else {
				int x = Integer.parseInt(st.nextToken());
				if (s.equals("add")) {
					set.add(x);
				} else if (s.equals("remove")) {
					set.remove(x);
				} else if (s.equals("check")) {
					bw.write(set.contains(x) ? "1\n" : "0\n");
				} else if (s.equals("toggle")) {
					if (set.contains(x)) {
						set.remove(x);
					} else {
						set.add(x);
					}
				}
			}
		}

		bw.flush();
	}
}
